app/models/health/qualifying_activity_v1.rb

```diff
@@ -87,77 +87,165 @@ def self.activities
           weight: 70,
+          allowed: ['U1', 'U2', 'U3', 'UK'],
+          required: [],
         },
         screening_completed: {
           title: 'Social services screening completed',
           code: 'T1023',
           weight: 80,
+          allowed: ['U1', 'U2', 'U3'],
+          required: [],
+          per_day: 1,
         },
         referral_to_aco: {
           title: 'Referral to ACO for Flexible Services',
           code: 'T1023>U6',
           weight: 90,
+          allowed: ['U1', 'U2', 'U3'],
+          required: ['U6'],
+          per_day: 1,
         },
         pctp_signed: {
           title: 'Person-Centered Treatment Plan signed',
           code: 'T2024>U4',
           weight: 100,
           hidden: true,
+          allowed: ['U1', 'U2', 'UK'],
+          required: ['U4'],
+          per_day: 1,
         },
         intake_completed: {
           title: 'Intake/Reassessment (completing consent/ROI, CHA, SSM, care plan)',
           code: 'G9005',
           weight: 110,
+          allowed: ['U1', 'U2', 'U3', 'UK'],
+          required: [],
         },
       }.sort_by { |_, m| m[:weight] }.to_h
     end
 
-    def contact_required?(activity)
-      return false unless activity
+    def modifiers
+      modifiers = []
+
+      # COVID rules permitted remote contacts to be treated as in-person,
+      # these rules ended with the end of the first CP program
+      contact_modifier = case @qa.activity&.to_sym
+      when :cha, :discharge_follow_up
+        if [:phone_call, :video_call].include?(@qa.mode_of_contact&.to_sym)
+          modes_of_contact[:in_person][:code]
+        else
+          modes_of_contact[@qa.mode_of_contact&.to_sym].try(:[], :code)
+        end
+      else
+        modes_of_contact[@qa.mode_of_contact&.to_sym].try(:[], :code)
+      end
+
+      # attach modifiers from activity
+      modifiers << activities[@qa.activity&.to_sym].try(:[], :code)&.split(/[ |>]/).try(:[], 1)
+
+      modifiers << contact_modifier
+      modifiers << client_reached[@qa.reached_client&.to_sym].try(:[], :code)
+
+      return modifiers.reject(&:blank?).compact
+    end
+
+    def internal_compute_valid_unpayable
+      reasons = []
+      computed_procedure_valid = @qa.compute_procedure_valid?
+
+      # Only valid procedures can be valid unpayable
+      return nil unless computed_procedure_valid
+
+      # Unpayable if it is a valid procedure, but it didn't occur during an enrollment
+      reasons << :outside_enrollment if computed_procedure_valid && ! @qa.occurred_during_any_enrollment?
+
+      # Unpayable if this was a phone/video call where the client wasn't reached
+      reasons << :call_not_reached if @qa.reached_client == 'no' && ['phone_call', 'video_call'].include?(@qa.mode_of_contact)
+
+      reasons << :limit_per_day unless @qa.within_per_day_limits?
+
+      # Signing a care plan is payable regardless of engagement status
+      return reasons if @qa.activity == 'pctp_signed'
```

Comment:
To match the return below.
```suggestion
      return reasons.uniq if @qa.activity == 'pctp_signed'
```